nimrod/report/report_analysis.py

```diff
@@ -0,0 +1,170 @@
+import re
+import os
+
+
+class Report_Analysis:
+
+    def __init__(self, randoop_original, randoop_modified):
+        self.randoop_original = randoop_original
+        self.randoop_modified = randoop_modified
+
+    def start_analysis(self):
+        left_randoop_original = self.randoop_original[0][2]
+        left_randoop_modified = self.randoop_modified[0][2]
+
+        if os.path.isdir(left_randoop_original) and os.path.isdir(left_randoop_modified):
+            self.suites_comparison(left_randoop_original, left_randoop_modified)
+            try:
+                right_randoop_original = self.randoop_original[4][2]
+                right_randoop_modified = self.randoop_modified[4][2]
+                self.suites_comparison(right_randoop_original, right_randoop_modified)
+            except:
+                print("It wasn't possible to compare the test suites for the commit: " + str(self.randoop_original[1][4]))
+        else:
+            print("It wasn't possible to compare the test suites for commits: " + str(self.randoop_original[0][4]) + " and " + str(self.randoop_original[1][4]))
+
+    def suites_comparison(self, path_suite_original, path_suite_modified):
+        methods = self.methods_report_comparison(path_suite_original, path_suite_modified)
+        objects = self.objects_report_comparison(path_suite_original, path_suite_modified)
+
+        if len(methods) == 0 or len(objects) == 0:
+            print("It wasn't possible to compare the suites " + str(path_suite_original) + " and " + str(path_suite_modified))
+        else:
+            self.write_comparison(path_suite_original, path_suite_modified, methods, objects)
+
+    def methods_report_comparison(self, path_suite_original, path_suite_modified):
+        methods_data = {}  # A dictionary in which the method name is the key and the value returned is a list with
+        # the number of times that the method was called for original randoop and randoop--
+
+        try:
+            methods_report_file = open(path_suite_original + "/methods_report.csv")
+            methods_report = methods_report_file.read()
+            methods_report_file.close()
+
+            lines = methods_report.split("\n")
+            for line in lines[1:-1]:  # Exclude the first one because contains the headers and the last one because it is a empty line
+                cells = re.split("(?<=\"),", line)
+                method_map = {cells[0]: [cells[1], 0]}
+                methods_data.update(method_map)
+        except:
+            print("Error on methods_report.csv in " + str(path_suite_original))
+            return {}
+
+        try:
```

Comment:
Lines 40-46 are duplicated in lines 54-60.
Besides that, lines 47-48 are duplicated in lines 64-65.
So I'd suggest extracting these lines as a new method.
This method might also receive the local variable `methods_data` as a parameter; this way, the lines 61-62 should also be part of this extracted method.